### PR TITLE
Corrected the code, passes seshat tests. Will need to check libseshat…

### PR DESCRIPTION
**Pull request by manderiasian**
… tests later

### CODE REVIEW

File: src/wrp_interface.c [hunk 1]
```diff
@@ -40,18 +40,26 @@ ssize_t wi_create_response_to_message(void *data, size_t cnt, void **message)
     wrp_msg_t response;
     ssize_t   message_size;
 
-    response.msg_type = WRP_MSG_TYPE__AUTH;
-    response.u.auth.status = 400;
+    memset(&response, 0, sizeof(wrp_msg_t));
+
     if( 0 < wrp_to_struct(data, cnt, WRP_BYTES, &in_msg) ) {
```
Comment (manderiasian) [inline src/wrp_interface.c]: It is there, it will return 400. Once this is merged I will look at it again.

File: src/wrp_interface.c [hunk 2]
```diff
@@ -85,8 +85,9 @@ ssize_t wi_create_response_to_message(void *data, size_t cnt, void **message)
 	    break;		
     
             default:
-            	return -1;    
-            break;
+            	response.msg_type = WRP_MSG_TYPE__AUTH;                 
+                response.u.auth.status = 501; // Not Implemented
```
Comment (cableramki) [inline src/wrp_interface.c]: From WEBPA-1573:- Unless a WRP_MSG_TYPE_REGISTRATION [9] or WRP_MSG_TYPERETRIEVE [6] is received a WRP type WRP_MSG_TYPE_AUTH [2] message with the status "400" shall be returned.

Comment (manderiasian) [inline src/wrp_interface.c]: I disagree, the specification needs to be modified. Error 501 would clearly indicate to the requester that "API Is Not Implemented.", vs 400 is a general error.

File: src/wrp_interface.c (hunk 1 above)
Comment (cableramki) [inline src/wrp_interface.c]: OK. :-)

File: src/wrp_interface.c (hunk 2 above)
Comment (cableramki) [inline src/wrp_interface.c]: OK. :-)

**Pull request merged**